frontend/src/lib/components/RadioSelect/index.tsx

```diff
@@ -9,23 +11,75 @@ export interface RadioSelectType {
 
 interface RadioSelectProps {
     options: RadioSelectType[]
-    selectedOption: null | string
-    onOptionChanged: (key: string | null) => void
+    selectedOption: null | string | string[]
+    onOptionChanged: (key: string | string[] | null) => void
+    multipleSelection?: boolean
+    focusSelection?: boolean // will hide other choices after making a selection
 }
 
-export function RadioSelect({ options, selectedOption, onOptionChanged }: RadioSelectProps): JSX.Element {
+export function RadioSelect({
+    options,
+    selectedOption,
+    onOptionChanged,
+    multipleSelection,
+    focusSelection,
+}: RadioSelectProps): JSX.Element {
+    const isSelected = (option: RadioSelectType): boolean => {
+        return multipleSelection && selectedOption
+            ? selectedOption?.includes(option.key)
+            : selectedOption === option.key
+    }
+
+    const handleClick = (option: RadioSelectType | null): void => {
+        if (!option) {
+            onOptionChanged(null)
+            return
+        }
+
+        if (multipleSelection) {
```

Comment:
:D This is quite messy - multi-selection with the same code tends to cause that.

But at the same time no direct business logic in here - good job!

Reply:
+1 agreed! thought about separating in two components but then the UI gets a bit messy, will keep it like this for now :)